src/local_cache.h

```diff
@@ -87,11 +125,46 @@ class LocalCache {
   // Request buffer from managed buffer
   TRITONSERVER_Error* Allocate(uint64_t byte_size, void** buffer);
 
+  // Cache Metric Helpers: note the metrics must be protected when accessed
+  TRITONSERVER_Error* InitMetrics();
+  TRITONSERVER_Error* UpdateMetrics();
+  // Returns the number of entries currently in cache
+  uint64_t NumEntries();
+  // Returns fraction of bytes allocated over total cache size between [0, 1]
+  double TotalUtilization();
+
+  // Cache Metrics: these must be protected when accessed
+  uint64_t num_evictions_ = 0;
+  uint64_t num_lookups_ = 0;
+  uint64_t num_hits_ = 0;
+  uint64_t num_misses_ = 0;
+  // Lookup/Insert latencies in microseconds
+  uint64_t total_lookup_latency_us_ = 0;
+  uint64_t total_insertion_latency_us_ = 0;
+  // TRITONSERVER_Metric/Family wrappers
+  TritonMetric cache_util_;
+  TritonMetric cache_entries_;
+  TritonMetric cache_hits_;
+  TritonMetric cache_misses_;
+  TritonMetric cache_lookups_;
+  TritonMetric cache_evictions_;
+  TritonMetric cache_lookup_time_;
+  TritonMetric cache_insert_time_;
+  // Thread to poll metrics at an interval
+  std::unique_ptr<std::thread> metrics_thread_;
+  std::atomic<bool> metrics_thread_exit_ = false;
+  // The interval at which metrics are updated by metric thread.
+  // Default interval is 1000ms = 1sec.
+  // NOTE: Can expose this as config field in the future
```

Comment:
I was wondering should this just use the `metrics-interval-ms` value that we have in the core?

Reply:
I thought of that originally, but we only expose a setter API for metrics interval (`TRITONSERVER_ServerOptionsSetMetricsInterval`), not a getter, and wasn't sure it was worth adding another API at this time unless requested to match. Based on user feedback/requests, the interval could be exposed to the cache config or the cache metrics could be changed to per-request instead of per-interval in the future which may make that getter API redundant.

Reply:
I think option (a) would be best. We can discuss this later.